CONTRIBUTING.md

```diff
@@ -0,0 +1,268 @@
+commit message, and there are some guidelines for how this should be formatted. Primarily, 
+
+* Keep the first line of the commit message 50 characters or less
+* If you need a longer description, keep the second line blank, and keep all subsequent more descriptive
+lines at 72 characters or less.
+
+The first line is what will be visible on the commit lists on GitHub, so make sure it's as descriptive as 
+you can make it.
+
+#### 5. Synchronizing your changes
+Sometimes, pull requests and code changes take time, and other contributions are accepted in the meantime.
+When this happens, you'll need to synchronize your changes with what's in the main repository. This should
+be done using `rebase`, not `merge`, to keep the commit history from being cluttered with merge commits.
+
+If you've not pushed your changes anywhere yet, it's sufficient to simply run (when on your branch)
+
+```bash
+$ git fetch upstream
+$ git rebase upstream/develop
+```
+
+to fetch the latest code and replay your work on it. However, if you've already pushed it, you might run
+into some issues when pushing to your fork after rebasing. To get around this, you'll have to forcibly push 
+your changes to overwrite what's in your repository.
+
+```bash
+$ git fetch --all
+$ git rebase upstream/develop
+$ git push --force-with-lease origin my-branch
+```
+
+#### 6. Opening a pull request
+When you feel that you're all done and you've pushed your changes to git, it's time to open a pull request
+and have your changes reviewed. Before doing so, run a final test by executing the build script in the 
+base directory of the codebase. 
+
+```bash
+$ ./build.sh
+```
+
+If it executes without any problems, you're good to go and ready to move on to creating your [Pull Request](#pull-requests).  
+
+## Pull Requests
+Pull requests are without a doubt one of the more involved contribution types. Primarily, in order for a
+pull request to be accepted, it must maintain a high quality, be well tested, and not have any major 
+breaking changes (unless absolutely neccesary). There's going to be a lot of stuff dumped on you in the
+next few paragraphs, but keep in mind that most are *guidelines*, not hard rules. Stick to them as best
+you can, and when in doubt - just ask.
+
+All pull requests must have or do the following:
+
+* A clear, concise and descriptive title. As a rule of thumb, don't make it longer than twelve words or 
+72 characters.
+* A clear and detailed description of what the pull request has changed. This includes how the behaviour
+of the library will change if the pull request is accepted - a maintainer should be able to read your 
+description and fully understand what accepting it would mean without having to dive into the code.
+* Be based on the `develop` branch of the main OpenTK repository.
+
+All pull requests should have the following: 
+* If applicable, a compilable example which demonstrates the changes. A git repository is preferred, and
```

Comment:
hmm we don't do this. Maybe an example to opentk-examples instead?

Reply:
This isn't referring to examples that should stick around - just a "this is what changed, and this is how it works now"

Reply:
ok